File: simulation/backend/CuttingMachine.java
public class CuttingMachine implements Machine {

    private float temperature, temperatureVariation, speed;

    //TODO: Add more arguments based on the factory.json
    public CuttingMachine(float temp, float tempVar, float speed){
        this.speed = speed;
        this.temperature = temp;
        this.temperatureVariation = tempVar;
    }

    @Override
    public void update() {
        this.temperature += this.temperatureVariation;
    }

    @Override
    //TODO: Add more information about the machine
    public String getRepresentation() {
        String representation = " - Temperature: " + temperature + "C\n";
        return representation;
    }
    
}

File: simulation/backend/CuttingMachine.js
export default class CuttingMachine {
  //TODO: Add more arguments based on the factory.json
  constructor(id, readingTime, links) {
    this.id = id;
    this.readingTime = readingTime;
    this.links = links;
    this.temperatureVariation = 0.1;
  }

  setVolt(volt){
    this.volt = volt;
  }
  
  setName(name){
    this.name = name;
  }

  setSpeed(speed){
    this.speed=speed;
  }
  
  setTemperature(temperature){
    this.temperature=temperature;
  }

  setPiecesProduced(piecesProduced){
    this.piecesProduced = piecesProduced;
  }
  
  setStatus(status){
    this.status=status;
  }
   
  setVibration(vibration){
    this.vibration = vibration;
  }
  
  setPressure(pressure){
    this.pressure = pressure;
  }
  
  setRotate(rotate){
    this.rotate = rotate;
  }

  update() {
    this.temperature += this.temperatureVariation;
  }

  //TODO: Add more information about the machine
  getRepresentation() {
    let representation = ` - Temperature:  ${this.temperature}C;`;
    return representation;
  }
}
